### PR TITLE
Upgrade to C# 10 [API-1091]

### PR DESCRIPTION
**Pull request by emreyigit**
No warnings occurred with the upgrade.

### CODE REVIEW

File: src/Hazelcast.Net/Hazelcast.Net.csproj [hunk 1]
```diff
@@ -5,6 +5,11 @@
     <RootNamespace>Hazelcast</RootNamespace>
   </PropertyGroup>
 
+  <PropertyGroup>
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net/Hazelcast.Net.csproj:8]: The default language version for all our projects is specified in `src/Directory.Build.Props` so I think you want to update that file, not any of the `.csproj` files. Other than that... if the PR triggers no new warning then it's all good.

**Review by zpqrtbnk — CHANGES_REQUESTED** [review]
See comment.

**Review by zpqrtbnk — APPROVED** [review]
Approved.

File: src/Hazelcast.Net/AssemblySigning.cs [hunk 2]
```diff
@@ -62,7 +62,9 @@ internal static string GetKeyToken(string key)
             var text = new StringBuilder();
             for (var i = 0; i < 8; i++)
                 //token[i] = hash[^(i + 1)];
+#pragma warning disable CA1305 // Specify IFormatProvider
```
Comment (srknzl) [inline src/Hazelcast.Net/AssemblySigning.cs:65]: Isn't this a warning?

**Review by srknzl — APPROVED** [review]
Approved but aren' there some warnings so you used pragma disables?

**Comment by emreyigit** [comment]
@srknzl they can be neglectable ones. Also, those part are mostly in test helpers.

**Pull request merged**